### PR TITLE
Add e2e test for upgrading management-components

### PR DESCRIPTION
**Pull request by d8660091**
*Issue #, if available:*

*Description of changes:*

*Testing (if applicable):* Manually invoked the new e2e test, and it passed.

*Documentation added/planned (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/e2e/docker_test.go [hunk 1]
```diff
@@ -1334,3 +1334,15 @@ func TestDockerKubernetesRegionalCuratedPackages(t *testing.T) {
...
+
+func TestDockerKubernetesUpgradeManagementComponents(t *testing.T) {
+	release := latestMinorRelease(t)
+	test := framework.NewClusterE2ETest(t, framework.NewDocker(t))
+
+	test.GenerateClusterConfig()
+	// create cluster with old eksa
+	test.CreateCluster(framework.ExecuteWithEksaRelease(release))
+	// upgrade management-components with new eksa
+	test.RunEKSA([]string{"upgrade", "management-components", "-f", test.ClusterConfigLocation, "-v", "99"})
```
Comment (cxbrowne1207) [inline test/e2e/docker_test.go:1346]: nit: I think that we should we create a `test.UpgradeManagmentComponents` function and hide these details.

**Review by cxbrowne1207 — COMMENTED** [review]
Do we also need this to test completing the rest of the upgrade after running upgrade management components?

**Comment by d8660091** [comment]
> Do we also need this to test completing the rest of the upgrade after running upgrade management components?

Because the followup procedure of upgrading the cluster itself will fail due to the bug: https://github.com/aws/eks-anywhere-internal/issues/2126. And I think right now we need a test case to cover the most essential flow of the new subcommand more than we need a comprehensive one that verify the flow of eksa controller reconciliation process. Having said that, we can do this in another PR when the bug is fixed

**Pull request merged**